internal/servers/plugin/v3/plugin.go

```diff
@@ -70,8 +70,10 @@ func (s *Server) Sync(req *pb.Sync_Request, stream pb.Plugin_SyncServer) error {
 	ctx := stream.Context()
 
 	syncOptions := plugin.SyncOptions{
-		Tables:     req.Tables,
-		SkipTables: req.SkipTables,
+		Tables:              req.Tables,
+		SkipTables:          req.SkipTables,
+		SkipDependentTables: req.SkipDependentTables,
+		DeterministicCQID:   req.DeterministicCqId,
```

Comment:
Actually, no, this needs to be passed as part of Init because it's passed on to the scheduler (unless we refactor the scheduler)

Reply:
I think we should rather change the scheduler though

Reply:
Refactored and updated this PR